File: content/java/v5.5/source/includes/crud/Delete.java
// Deletes documents from a collection by using the Java driver

package org.example;

import static com.mongodb.client.model.Filters.eq;

import org.bson.Document;
import org.bson.conversions.Bson;

import com.mongodb.MongoException;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.result.DeleteResult;

import static com.mongodb.client.model.Filters.lt;

public class Delete {

    public static void main(String[] args) {
        // Replace the uri string with your MongoDB deployment's connection string
        String uri = "<connection string uri>";

          try (MongoClient mongoClient = MongoClients.create(uri)) {

            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");

            Bson deleteOneQuery = eq("title", "The Garbage Pail Kids Movie");

            // Deletes the first document that has a "title" value of "The Garbage Pail Kids Movie"
            DeleteResult result = collection.deleteOne(deleteOneQuery);
            System.out.println("Deleted document count - delete one: " + result.getDeletedCount());

            Bson deleteManyQuery = lt("imdb.rating", 1.9);

            // Deletes all documents that have an "imdb.rating" value less than 1.9
            result = collection.deleteMany(deleteManyQuery);

            // Prints the number of deleted documents
            System.out.println("Deleted document count - delete many: " + result.getDeletedCount());
        }
    }
}

File: v5.2/examples/src/main/kotlin/usageExamples/delete/delete.kt
package usageExamples.delete

// :replace-start: {
//    "terms": {
//       "CONNECTION_URI_PLACEHOLDER": "\"<connection string uri>\"",
//       "import io.github.cdimascio.dotenv.dotenv\n": ""
//    }
// }
// :snippet-start: delete-usage-example

import com.mongodb.MongoException
import com.mongodb.client.model.Filters
import com.mongodb.kotlin.client.coroutine.MongoClient
import io.github.cdimascio.dotenv.dotenv
import kotlinx.coroutines.runBlocking


data class Movie(val title: String)

fun main() = runBlocking {
    // :remove-start:
    val dotenv = dotenv()
    val CONNECTION_URI_PLACEHOLDER = dotenv["MONGODB_CONNECTION_URI"]
    // :remove-end:
    // Replace the uri string with your MongoDB deployment's connection string
    val uri = CONNECTION_URI_PLACEHOLDER
    val mongoClient = MongoClient.create(uri)
    val database = mongoClient.getDatabase("sample_mflix")
    val collection = database.getCollection<Movie>("movies")

    collection.insertOne(Movie("The Garbage Pail Kids Movie")) // :remove:
    val query = Filters.eq(Movie::title.name, "The Garbage Pail Kids Movie")

    try {
        val result = collection.deleteOne(query)
        println("Deleted document count: " + result.deletedCount)
    } catch (e: MongoException) {
        System.err.println("Unable to delete due to an error: $e")
    }
    // :remove-start:
    // clean up
    database.drop()
    // :remove-end:
    mongoClient.close()
}
// :snippet-end:
// :replace-end: